### PR TITLE
Support binding to return values

### PR DESCRIPTION
**Pull request by mathewc**
Addresses https://github.com/Azure/azure-webjobs-sdk-script/issues/675

Still adding more tests. I'll also go in and simplify some of the other samples. I'm also going to look into rationalizing our existing special case return handling for http response with this.

### CODE REVIEW

**Comment by brettsam** [comment]
Awesome! I know you're still adding tests -- make sure to add at least one e2e that handles the 'mixed' mode where you have a return and an 'out' param.

File: src/WebJobs.Script/Description/Node/NodeFunctionInvoker.cs [hunk 1]
```diff
@@ -172,15 +172,26 @@ static NodeFunctionInvoker()
                 return;
             }
 
-            // if the function returned binding values via the function result,
-            // apply them to context.bindings
             var bindings = (Dictionary<string, object>)scriptExecutionContext["bindings"];
-            IDictionary<string, object> functionOutputs = functionResult as IDictionary<string, object>;
-            if (functionOutputs != null)
+            var returnValueBinding = outputBindings.SingleOrDefault(p => string.Compare(ScriptConstants.SystemReturnParameterBindingName, p.Metadata.Name) == 0);
```
Comment (fabiocav) [inline src/WebJobs.Script/Description/Node/NodeFunctionInvoker.cs]: Would be nice to use the metadata to match on the `IsReturn` property so the logic is consistent across invokers.

File: src/WebJobs.Script/Description/DotNet/DotNetFunctionDescriptorProvider.cs [hunk 2]
```diff
@@ -180,5 +190,36 @@ protected override IFunctionInvoker CreateFunctionInvoker(string scriptFilePath,
...
+        {
+            descriptor = null;
+
+            var returnBinding = bindings.SingleOrDefault(p => p.Metadata.IsReturn);
+            if (returnBinding == null)
+            {
+                return false;
+            }
+
+            if (functionReturnType.IsGenericType && functionReturnType.GetGenericTypeDefinition() == typeof(Task<>))
```
Comment (fabiocav) [inline src/WebJobs.Script/Description/DotNet/DotNetFunctionDescriptorProvider.cs]: Should the check here be `functionReturnType is Task` instead? Then check if is generic? We may also want to consider failing with a descriptive message if we have a non-generic task.

File: src/WebJobs.Script/Description/Node/NodeFunctionInvoker.cs (hunk 1 above)
Comment (mathewc) [inline src/WebJobs.Script/Description/Node/NodeFunctionInvoker.cs]: Yep, missed this one :)

File: src/WebJobs.Script/Description/DotNet/DotNetFunctionDescriptorProvider.cs (hunk 2 above)
Comment (mathewc) [inline src/WebJobs.Script/Description/DotNet/DotNetFunctionDescriptorProvider.cs]: Done. Also added test.

**Comment by mathewc** [comment]
@brettsam I added some mixed mode tests for Node and C#

**Comment by mathewc** [comment]
Merged, thanks guys.

**Pull request closed without merging**